src/content.js

```diff
@@ -118,45 +127,55 @@ async function getPageMetadata(contentDir) {
   return pages;
 }
 
-async function buildMetaIndex(contentDir, invaderDefsDir, baseUrl, packageVersion) {
```

Comment:
The `metaIndex` API is no longer constructed for use by all pages, but is now a per-page object. Importantly, the function `resolvePage` needs to have context on the page being rendered so it can consider which destination is more related. You'll see this code is moved down to within `renderContent`'s page loop.